dataplane/standalone/saiserver/hostif.go

```diff
@@ -24,21 +24,110 @@ import (
 	"google.golang.org/protobuf/proto"
 
 	"github.com/openconfig/lemming/dataplane/forwarding/fwdconfig"
-	saipb "github.com/openconfig/lemming/dataplane/standalone/proto"
+	"github.com/openconfig/lemming/dataplane/internal/engine"
+	"github.com/openconfig/lemming/dataplane/standalone/cpusink/sink"
 	"github.com/openconfig/lemming/dataplane/standalone/saiserver/attrmgr"
+
+	saipb "github.com/openconfig/lemming/dataplane/standalone/proto"
 	dpb "github.com/openconfig/lemming/proto/dataplane"
 	fwdpb "github.com/openconfig/lemming/proto/forwarding"
 )
 
-func newHostif(mgr *attrmgr.AttrMgr, dataplane switchDataplaneAPI, s *grpc.Server) *hostif {
-	p := &hostif{
+func newHostif(ctx context.Context, mgr *attrmgr.AttrMgr, dataplane switchDataplaneAPI, s *grpc.Server) (*hostif, error) {
+	hostif := &hostif{
 		mgr:              mgr,
 		dataplane:        dataplane,
 		trapIDToHostifID: map[uint64]uint64{},
 		groupIDToQueue:   map[uint64]uint32{},
 	}
-	saipb.RegisterHostifServer(s, p)
-	return p
+
+	// Setup the packet io tables. A packet is punted by setting the output port to the CPU port.
+	// There a two places where packets can be punted:
+	//   1. pre-fib: the trap table contains rules that may punt the packets.
+	//   2. fib: "IP2ME" routes, the fib contains routes for the IPs assigned to the interface, these routes have the next hop as the CPU port.
+	// Once a packet is sent to the CPU port, it must be matches to a host interfaces:
+	//   1. ip2me: a table maps IP DST to hostif port. (populated by the CPU port).
+	//   2. hostif table: a table the maps TRAP IP to hostif port. (trap id is set by the ACL actions).
```

Comment:
Can I ask a basic question: what the difference between host interfaces and regular interfaces?

Reply:
updated this comment to more consistent with wording
there are two types:
1. port: physical port on a device (in KNE world this is one end of the veth that connect nodes)
2. hostif: linux network dev that represents the port to the OS (in lemming this TAP interfaces in the pod)